### ISSUE TITLE
Cannot set default latitude/longitude in MapBox chart

### ISSUE THREAD

**Issue by frafra** [apache/superset#17269]
#### How to reproduce the bug

1. Create a new Chart
2. Select MapBox
3. Select latitude and longitude
4. Click on Customize
5. Zoom and pan on the map

### Expected results

Change viewport default parameter.

### Actual results

Latitude and longitude do not change, while the zoom parameter is glitchy.

### Environment

- browser type and version: Firefox 93
- superset version: `superset version` the latest Superset from Docker

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by srinify** [comment]
Hi @frafra what would help here is:

- Specific version of Superset you're using (or git sha)
- Screenshots of the issue
- Any errors in the browser console you're seeing

Without this, it's going to be quite hard for anyone here to help out.

**Comment by frafra** [comment]
@srinify it looked easy to reproduce to me (just try to set a different latitude on a map, nothing happens), but I will send a video as requested.

**Comment by frafra** [comment]
https://user-images.githubusercontent.com/4068/145795379-33d47264-6afd-4e2f-b847-13abde9651a6.mp4

As you can see, panning does not change latitude/longitude. Writing custom coordinates does not have any effect, even if the widget is saved and reloaded. The zoom glitches when zooming in.

**Comment by beersweetpicklepanda** [comment]
Can confirmed as well, Docker Superset 1.3.0

**Comment by frafra** [comment]
Still valid.

**Comment by JakobMiksch** [comment]
I can reproduce it as well

**Comment by JakobMiksch** [comment]
There seems to be a similar issue with the Deck.GL plugin. The popover of the Viewport-control does not properly update either.
![image](https://user-images.githubusercontent.com/15704517/203534282-9a17afe6-cbd1-469f-9448-5ba4364cbbcd.png)

**Comment by JohnDietrich-Pepper** [comment]
Yep this is an issue in Deck.Gl as well.  Would be nice to be able to set the default view.

**Comment by jseparovic** [comment]
The lat/long/zoom values are not being used as the initial set of values when the component loads. Instead the bounds are used to calculate the viewport.
I've added a control setting to switch from this behavior "Auto" to using "Fixed" values based on the lat/long/zoom in the viewport settings.
![image](https://github.com/apache/superset/assets/4665758/13434f5c-3f4c-4b80-a18f-fda61371201d)

I was also thinking that the setControlValue call should be saving the viewport settings when the map is dragged, so I thought an additional settings called "User" would be good to remember the User's dragged lat/long/zoom. However, it wasn't quite working as I expected and didn't seem to update the viewport settings properly.

I also noticed that on first load after login, the plots show up ok on the map, but if you click to another dashboard and then back again, the plots do not show up until you drag the map. This was due to clusters not being tracked in react state. I updated the component to an FC and fixed this issue by add clusters to useState.

There's still one more issue when looking at the world map and scanning left or right. When a plot disappears out of screen and then the same location shows up on the other side of the screen, the plot does not appear as soon as the location is visible.

Here is the map I'm expecting:
![image](https://github.com/apache/superset/assets/4665758/8ae919d0-3cd6-4287-9d06-b5c3ed46409c)

After dragging to the left:
![image](https://github.com/apache/superset/assets/4665758/26ad6610-e54c-4079-9866-9f3fb84d32cb)

Looks like the plot calculation needs to be updated to fix this issue. Possibly to create duplicate plots so if the location is visible twice on the map, the plot shows up twice. This can happen when zooming all the way out.

**Comment by rusackas** [comment]
I assume folks are still facing this... code reviews on the linked PR are more than welcome :)

**Comment by michael-s-molina** [comment]
Thank you for reporting this issue. However, the issue was reported on a version of Superset that we no longer support or it does not contain a valid Superset version. As of this moment, we only actively support Superset 4.0 and 4.1. To maintain a more actionable Issues backlog, we're going to close this issue for now. If you (or anyone reading this) are still experiencing this on a currently supported version of Superset, please either reopen this Issue, or file a new one with updated context (screenshots, reproduction steps) and we'll do our best to support it. Thank you
